src/manage.c

```diff
@@ -5413,6 +5428,130 @@ manage_sync (sigset_t *sigmask_current,
     }
 }
 
+/**
+ * @brief Perform any processing of imported reports that is due.
+ *
+ * In gvmd, periodically called from the main daemon loop.
+ */
+void
+manage_process_report_imports ()
+{
+  lockfile_t lockfile;
+  iterator_t reports;
+  report_t report;
+  int pid, ret;
+
+  reinit_manage_process ();
+  manage_session_init (current_credentials.uuid);
+
+  init_iterator (&reports, "SELECT id FROM reports"
```

Comment:
Too much for me to review all of this, but SQL should go in `manage_sql.c` (or some other `*_sql.c`). One option would be to add add a new iterator, see for example `init_report_iterator_task`.